### DISCUSSION TITLE
Having a hard time adjusting to Geany from EditPlus

### DISCUSSION THREAD

**Discussion by m1kemex** [geany/geany#3793] [Q&A]
Hi,

I used EditPlus for coding years ago. It's really a great program, to be honest. But I've always tried to support the Free Software movement by at least trying to use as much GPL applications as possible. Therefore, when I resumed programming, I went to see if there was a free equivalent to EditPlus. And I was glad to find Geany.

However, I'm having a really hard time adjusting to it. Maybe it just does things differently and I'm simply not used to it, but I find that even simple tasks can't be accomplished. I just filed a bug report because I have a very large UTF-8 encoded file (about 18MB) and I can't convert it to ISO-8859-1 without throwing an error it shouldn't produce in the first place (I checked the file with an hex editor and encoding seems valid).

Now I'm having a really hard time using regular expressions to process an SQL file. In EditPlus I could do simple things like replace a duplicated line (replace '\n\n' with '\n') but Geany "doesn't find anything". I've tried both regular mode and multi-line. And I can't seem to make it to work.

Given the similarities between EditPlus and Geany, I'd like to suggest some more changes. For now, I'd be happy with having it to run properly.

**Comment by elextr** [comment]
> In EditPlus I could do simple things like replace a duplicated line (replace '\n\n' with '\n') but Geany "doesn't find anything"

Search for: `\n\n` and replace with `\n` and both `use regular expressions` and `use multi-line matching` checked works for me.  Is your file an olde windows one with CRLF not LF as line ends?

> However, I'm having a really hard time adjusting to it. Maybe it just does things differently and I'm simply not used to it, but I find that even simple tasks can't be accomplished. I just filed a bug report because I have a very large UTF-8 encoded file (about 18MB) and I can't convert it to ISO-8859-1 without throwing an error it shouldn't produce in the first place (I checked the file with an hex editor and encoding seems valid).

See that issue for where you were wrong.

> Given the similarities between EditPlus and Geany, I'd like to suggest some more changes. For now, I'd be happy with having it to run properly.

Geany is not EditPlus, it is not identical, and it does "run properly", but not the same.  Different is not improper.
